packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -107,9 +107,36 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
       this.startupTimeout
     );
 
+    await Promise.race([this.waitContainerNotExited(container), healthCheckPromise]);
+
     log.debug(`HTTP wait strategy complete`, { containerId: container.id });
   }
 
+  private async waitContainerNotExited(container: Dockerode.Container) {
```

Comment:
I will write tests for it if you approve me this idea @cristianrgreco

Reply:
I like the idea @NuShoSinkuPomogliTebeTvoiSankcii, thanks for raising the PR. 

Yes please do write a test for this, I _think_ you'll encounter an issue where the test doesn't terminate, because although the waitContainerNotExited promise has completed, the healthCheckPromise is still running. If this is the case, I think you could put a condition in the healthcheck promise to see if the container has already exited and stop the interval if that's the case.

Reply:
Thanks for your point! I wrongly thought that Promise.race close other promises after any promise finished.
Moved status check inside existing IntervalRetry.